### ISSUE TITLE
object destructuring assignment should have a dedicated option for the [typedef] rule

### ISSUE THREAD

**Issue by drywolf** [palantir/tslint#1039]
Currently when using a object-destructuring statement, tslint handles it just as a regular variable declaration.

![image](https://cloud.githubusercontent.com/assets/394294/13820238/ba44b07c-eb9c-11e5-8c43-e0bff523b6a6.png)

It would be a good feature to let this es2015 construct have it's own separate option in the "typedef" rule.
(the same also applies for [array-destructuring expressions](http://www.2ality.com/2015/01/es6-destructuring.html))

For example this would be a config with the new options:

        "typedef": [
            true,
            "call-signature",
            "parameter",
            "property-declaration",
            "variable-declaration",
            "member-variable-declaration",
            // those are new
            "object-destructuring",
            "array-destructuring"
        ],

Regards

**Comment by JKillian** [comment]
Good idea! I think explicit types for this kind of construct is pretty rare. Nobody wants to be writing types like this:

```ts
const obj = { foo: "bar" };
const { foo }: { foo: string } = obj;
```
